src/agent/tests/unix_daemon_test.cpp

```diff
@@ -2,26 +2,41 @@
 
 #include <unix_daemon.hpp>
 
+#include <filesystem>
+#include <fstream>
+
 class UnixDaemonTest : public ::testing::Test
 {
 protected:
-    void SetUp() override {}
+    std::filesystem::path m_tempConfigFilePath;
+
+    void SetUp() override
+    {
+        m_tempConfigFilePath = "temp_wazuh-agent.yml";
+
+        std::ofstream outFile(m_tempConfigFilePath);
+        outFile << R"(
+            agent:
+                path.run: "."
+        )";
+        outFile.close();
+    }
```

Comment:
Isn't it possible to only create this file once? The `SetUp` method is called in between every test in the fixture. You could also do it in the constructor of `UnixDaemonTest`.

Who deletes this file?

Reply:
Fixed! I replace the functions SetUp() and TearDown() with SetUpTestSuite() and TearDownTestSuite().